src/napari_matplotlib/histogram.py

```diff
@@ -65,20 +70,25 @@ def draw(self) -> None:
 
         # Important to calculate bins after slicing 3D data, to avoid reading
         # whole cube into memory.
-        bins = np.linspace(np.min(data), np.max(data), 100)
+        if data.dtype.kind in {"i", "u"}:
+            # Make sure integer data types have integer sized bins
+            step = 1 + abs(np.max(data) - np.min(data)) // 100
```

Comment:
we also need to use integer sized bins for images with unsigned ints, and take the `abs` of the difference in case all elements are less than or equal to 0 (otherwise the step size would be negative)

Reply:
I've updated the baseline images as previously float bins were still being used (because the dtype kind is `u` rather than `i`)